src/js/profile/subindicator_filter.js

```diff
@@ -122,51 +126,44 @@ export class SubindicatorFilter extends Observable {
         $(dropdown).find('.dropdown-menu__selected-item .truncate').text(selected);
         $(dropdown).find('.dropdown-menu__content').hide();
 
-        if (callback !== null) {
-            callback(selected);
+        if (callback !== null && item !== null) {
+            callback(item);
         }
     }
 
     groupSelected = (selectedGroup, subindicatorDd, title) => {
         let callback = (selectedFilter) => this.parent.applyFilter(this.getFilteredData(selectedFilter, selectedGroup, title), selectedGroup, selectedFilter);
         this.resetDropdown(subindicatorDd);
-        callback(ALLVALUES);
+        //callback(ALLVALUES);
         if (selectedGroup !== ALLVALUES) {
-          let subindicators = Object.keys(this.indicators[title].groups[selectedGroup]);
-          this.populateDropdown(subindicatorDd, subindicators, callback);
+            let subindicators = selectedGroup.subindicators;
+            this.populateDropdown(subindicatorDd, subindicators, callback);
         }
     }
 
-    getFilteredGroups(groups, selectedGroup, selectedFilter) {
-        const group = groups[selectedGroup]
-        if (group == undefined)
-            return []
-
-        const groupValue = Object.entries(group).find(g => g[0] == selectedFilter)
-        if (groupValue == undefined)
-            return []
-
-        const subindicators = Object.entries(groupValue[1])
-        return subindicators.map(cd => new SubIndicator(cd))
-    }
+    getFilteredData = (selectedFilter, selectedGroup, title) => {
+        this.triggerEvent('point_tray.subindicator_filter.filter', {
+            indicator: title,
+            group: selectedGroup,
+            subindicator: selectedFilter
+        });
 
-    getFilteredSubindicators(subindicators) {
-        return subindicators.map(el => {
-            return new SubIndicator([el.label, el])
-        })
+        if (selectedFilter !== ALLVALUES)
+            return this.getFilteredGroups(selectedGroup, selectedFilter)
+        else
+            return this.getFilteredSubindicators(subindicatorData.subindicators)
```

Comment:
what is `subindicatorData` ? and `subindictors´ in that context?

Reply:
this is updated - in `else` block we return `this.childData` now